bundles/org.openhab.binding.visualcrossing/src/main/java/org/openhab/binding/visualcrossing/internal/VisualCrossingHandler.java

```diff
@@ -0,0 +1,554 @@
+                    }
+                    if (dayHour.hourUvIndex().equals(channelId)) {
+                        return newDecimalType(findHour(hours, hourIdx), Hour::uvindex);
+                    }
+                    if (dayHour.hourSevereRisk().equals(channelId)) {
+                        return newDecimalType(findHour(hours, hourIdx), Hour::severerisk);
+                    }
+                    if (dayHour.hourConditions().equals(channelId)) {
+                        return newStringType(findHour(hours, hourIdx), Hour::conditions);
+                    }
+                    if (dayHour.hourIcon().equals(channelId)) {
+                        return newStringType(findHour(hours, hourIdx), Hour::icon);
+                    }
+                    if (dayHour.hourStations().equals(channelId)) {
+                        return newStringCollectionType(findHour(hours, hourIdx), Hour::stations);
+                    }
+                    if (dayHour.hourSource().equals(channelId)) {
+                        return newStringType(findHour(hours, hourIdx), Hour::source);
+                    }
+                }
+            }
+        }
+
+        logger.warn("Do not know this channel ID: {}", channelId);
+        return UNDEF;
+    }
+
+    @Nullable
+    private Hour findHour(List<Hour> hours, int hourIdx) {
+        if (hourIdx >= hours.size()) {
+            return null;
+        }
+        return hours.get(hourIdx);
+    }
+
+    @Override
+    public void initialize() {
+        updateStatus(OFFLINE, CONFIGURATION_PENDING, "@text/channel-type.visualcrossing.weather.config-pending");
+        weatherResponse.set(null);
+
+        var config = getConfigAs(VisualCrossingConfiguration.class);
+        var hostname = config.hostname;
+        try {
+            new URL(hostname);
+        } catch (MalformedURLException e) {
+            logger.debug("Hostname [{}] is not an URL!", hostname, e);
+            updateStatus(OFFLINE, CONFIGURATION_ERROR,
+                    "@text/addon.visualcrossing.weather.error.hostname [\"%s\"]".formatted(hostname));
+            return;
+        }
+
+        var apiKey = config.apiKey;
+        if (apiKey == null) {
+            updateStatus(OFFLINE, CONFIGURATION_ERROR, "@text/addon.visualcrossing.weather.error.missing-api-key");
+            return;
+        }
+
+        var restClient = (RestClient) new HttpClient(httpClientFactory.getCommonHttpClient());
+        if (config.httpRetries > 0) {
+            restClient = new RetryHttpClient(restClient, config.httpRetries);
+        }
+
+        if (config.location != null && !config.location.isBlank()) {
+            location = config.location;
+        } else {
+            // todo location from OH
+        }
+
+        if (config.lang != null && !config.lang.isBlank()) {
+            lang = config.lang;
+        } else {
+            // todo location from OH
```

Comment:
@lsiepel do you know how can I get language that is set in MainUI for OpenHAB?

Reply:
Same as previous, check https://www.openhab.org/javadoc/latest/org/openhab/core/i18n/package-summary